src/ol/featureloader.js

```diff
@@ -130,25 +130,38 @@ export function loadFeaturesXhr(url, format, success, failure) {
  * @api
  */
 export function xhr(url, format) {
-  return loadFeaturesXhr(
-    url,
-    format,
-    /**
-     * @param {Array<import("./Feature.js").default>} features The loaded features.
-     * @param {import("./proj/Projection.js").default} dataProjection Data
-     * projection.
-     * @this {import("./source/Vector").default|import("./VectorTile.js").default}
-     */
-    function (features, dataProjection) {
-      const sourceOrTile = /** @type {?} */ (this);
-      if (typeof sourceOrTile.addFeatures === 'function') {
-        /** @type {import("./source/Vector").default} */ (sourceOrTile).addFeatures(
-          features
-        );
-      }
-    },
-    /* FIXME handle error */ VOID
-  );
+  /**
+   * @param {import("./extent.js").Extent} extent Extent.
+   * @param {number} resolution Resolution.
+   * @param {import("./proj/Projection.js").default} projection Projection.
+   * @param {function(): void=} success Success
+   *      Function called when loading succeeded.
+   * @param {function(): void=} failure Failure
+   *      Function called when loading failed.
+   * @this {import("./source/Vector").default}
+   */
+  return function (extent, resolution, projection, success, failure) {
+    const sourceOrTile = /** @type {import("./source/Vector").default} */ (this);
+    loadFeaturesXhr(
+      url,
+      format,
+      extent,
+      resolution,
+      projection,
+      /**
+       * @param {Array<import("./Feature.js").default>} features The loaded features.
+       * @param {import("./proj/Projection.js").default} dataProjection Data
+       * projection.
+       */
+      function (features, dataProjection) {
+        if (success !== undefined) {
+          success();
+        }
+        sourceOrTile.addFeatures(features);
```

Comment:
You'd still have to check for availabilty of `addFeatures`, because vector tiles don't have that function. But I have a better idea: Now that you have a `success` callback, this function could be improved if it wouldn't deal with adding features at all any more. Then the tile or source would take care of adding features in the `success` callback. Should be a straightforward change, if you change the `success` callback to accept `features` as argument.

Reply:
The xhr function is not actually used in the vector tiles. There exists a `defaultLoadFunction` which does this. That's why I proposed in https://github.com/openlayers/openlayers/issues/9975#issuecomment-704092256 that the functionality could be moved into a similar default function in the `source/Vector` file.

I agree that this could be done with a callback as well. I abstained from doing it, because it would break existing custom load functions. But if that's ok, I can do that.

Reply:
Let's not break existing custom load function. Good point!

Thanks for the explanations! I had missed that the function is not used by the VectorTile source any more. I don't think it's necessary to do the refactoring you proposed in https://github.com/openlayers/openlayers/issues/9975#issuecomment-704092256. I think the only thing that's left to do here is rename the `sourceOrTile` variable to `source`, and change its type accordingly.